### PR TITLE
[Motion] Use rootMargin instread of threshold to trigger slide in animations

### PR DESCRIPTION
**Pull request by ludoboludo**
### PR Summary: 

Tweak the approach we use to trigger the animations

### Why are these changes introduced?

Since we're currently relying on a the threshold value to tell when to trigger the animations, it's a percentage value of the element in question. So when it appears will vary based on that element's height. 

### What approach did you take?

By relying on rootMargin instead we apply a consistent trigger to our animations 👍 

### Visual impact on existing themes

Should trigger the appearance of upcoming sections more regularly. 

### Testing steps/scenarios

- [ ] Enable the animation in the general settings and make sure each section is getting animated as expected

### Demo links

- [Editor](https://os2-demo.myshopify.com/admin/themes/139694145558/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by kjellr** [comment]
I like this! It seems to work pretty well in my testing — the items all animate in at the same point, regardless of their height. The `rootMargin` value also seems spot on to me. 🌟 

Here's a small weird thing I wonder if we can clear up as a special case (either here, or in another PR): Check out this example: 

https://user-images.githubusercontent.com/1202812/231467141-d0bba8c5-98fb-4a01-bfc1-dda9a81b681f.mp4

As you can see, the body text for the Rich Text block doesn't appear on initial load — it only kicks in when you scroll. This feels wrong. Ideally, when something is even a little above the fold, we'd animate it in on load. 

I think what we want is: 

- If something's visible in the viewport on initial load, animate it in. 
- Otherwise, set the `rootMargin` value you have here. 

What do you think?

File: assets/animations.js [hunk 1]
```diff
@@ -18,7 +18,8 @@ function initializeScrollAnimationTrigger(rootEl = document) {
   if (animationTriggerElements.length === 0) return;
 
   const observer = new IntersectionObserver(onIntersection, {
-    threshold: 0.1,
+    rootMargin: '0px 0px -50px 0px',
```
[the remark this answers is not in the document]
Comment (kjellr) [reply assets/animations.js:21]: If it seems difficult to achieve, I'm perfectly happy merging this in for now and addressing it in a followup.

[the remark this answers is not in the document]
Comment (kjellr) [reply assets/animations.js:21]: If it seems difficult to achieve, I'm perfectly happy merging this in for now and addressing it in a followup.

File: assets/animations.js [hunk 2]
```diff
@@ -18,7 +18,8 @@ function initializeScrollAnimationTrigger(rootEl = document) {
...
+    threshold: 0,
```
Comment (metamoni) [inline assets/animations.js:22]: This is the default value, so if we're leaving it at 0 you can remove it

File: assets/animations.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ludoboludo) [reply assets/animations.js:21]: Yeah I think I will leave it as is for now. And see if it comes up often as an issue in our testing. We could reduce the negative root margin value if needed.

**Pull request merged**